discretenet/problems/gisp/gisp.py

```diff
@@ -0,0 +1,216 @@
+            coeffs[node] = self.graph.nodes[node]["revenue"]
+
+        model.E2 = pyo.Set(initialize=self.E2)
+        model.y = pyo.Var(model.E2, domain=pyo.Binary)
+        for node1, node2 in self.E2:
+            # model.y[node1, node2] = pyo.Var(domain=pyo.Binary)
+            coeffs[node1, node2] = -1 * self.graph[node1][node2]["cost"]
+
+        #  objective function
+        model.objective = pyo.Objective(
+            expr=pyo.quicksum(
+                coeffs[node] * model.x[node] for node in self.graph.nodes()
+            )
+            + pyo.quicksum(
+                coeffs[node1, node2] * model.y[node1, node2] for node1, node2 in self.E2
+            ),
+            sense=pyo.maximize,
+        )
+
+        #  add constraints
+        model.constraints = pyo.ConstraintList()
+        for node1, node2, edge in self.graph.edges(data=True):
+            if (node1, node2) not in self.E2:
+                model.constraints.add(model.x[node1] + model.x[node2] <= 1)
+            else:
+                model.constraints.add(
+                    model.x[node1] + model.x[node2] - model.y[node1, node2] <= 1
+                )
+
+        self.model = model
+
+    def is_linear(self) -> bool:
+        return True
+
+    def get_name(self) -> str:
+        return self.name
+
+
+class GISPGenerator(Generator):
+    def __init__(
+        self,
+        random_seed: int = 42,
+        path_prefix: Union[str, Path] = None,
+        which_set="SET2",
+        graph_instance=None,
+        min_n=100,
+        max_n=100,
+        er_prob=0.1,
+        set_param=100.0,
+        alpha=0.75,
+    ):
+        """
+        Initialize the GISP generator instance following https://doi.org/10.1016/j.ejor.2016.11.050
+        :param random_seed: The random seed to use
+        :param path_prefix:  Path prefix to pass to instance ``save()`` methods
+            during batch generation. Must be set as a public instance attribute,
+            to be changed by the user after generator instantiation if desired.
+        :param which_set: can take values “SET1” or “SET2". These are two types
+            of models for assigning costs/revenues to edges/nodes in the generated
+            instances.
+        :param graph_instance: path to .clq file of DIMACS graph or do not provide
+            it and a random graph will be generated.
+        :param min_n/max_n: min/max number of nodes a randomly generated
+            graph can have, if not using a DIMACS graph
+        :param er_prob: the probability of an edge existing in the randomly
+            generated graph, if not using a DIMACS graph.
+        :param set_param: a value that affects the node revenues and/or edge costs
+        :param alpha: probability of an edge being in the set of edges which are
+            legal to remove (at a cost)
+        """
+
+        super().__init__(random_seed, path_prefix)
+        random.seed(random_seed)  # should be moved into set_seed()
```

Comment:
Feel free to do that in the base `set_seed()` method in `Problem`

Reply:
Updated